internal/websocket/ws.go

```diff
@@ -152,13 +155,18 @@ func (ws *Websocket) Start() {
 
 	var ctx context.Context
 	ctx, ws.shutdown = context.WithCancel(context.Background())
+	// Init retry policy, but it'll be reset on recurring successful connections.
+	ws.policy = ws.retryPolicyFactory.NewPolicy(ctx)
+	ws.decoder = wrp.NewDecoder(nil, wrp.Msgpack)
 
-	go ws.run(ctx)
+	go ws.readPump(ctx)
 }
 
 // Stop stops the websocket connection.
 func (ws *Websocket) Stop() {
 	ws.m.Lock()
+	// Avoid the overhead of the close handshake.
+	ws.conn.Close(nhws.StatusNormalClosure, "")
```

Comment:
I'm seeing reader errors when we close the socket with nhws.StatusNormalClosure
```
Received unexpected error:
        	            	failed to get reader: received close frame: status = StatusNormalClosure and reason = ""
```

Reply:
I would assume `nhws.StatusNormalClosure` would trigger a `io.EOF` and not a unexpected error

Reply:
looks like nhooyr.io/websocket's `Reader` doesn't return an `io.EOF` for a `StatusNormalClosure` (status code 1000).
https://github.com/xmidt-org/xmidt-agent/pull/57#discussion_r1548670211

just a fyi